runtime/src/qio/regex/bundled/re2-interface.cc

```diff
@@ -244,10 +244,15 @@ void qio_regex_get_options(const qio_regex_t* regex, qio_regex_options_t* option
   re2_options_to_qio_re_options(&opts, options);
 }
 
-void qio_regex_get_pattern(const qio_regex_t* regex, const char** pattern)
+void qio_regex_get_pattern(const qio_regex_t* regex, const char** pattern, int64_t* len_out)
 {
   RE2* re2 = (RE2*) regex->regex;
-  *pattern = qio_strdup(re2->pattern().c_str());
+  const std::string &s = re2->pattern();
+  char *copy = (char*) qio_malloc(s.length()+1);
+  memcpy(copy, s.data(), s.length());
+  copy[s.length()] = '\0';
+  *pattern = copy;
+  *len_out = re2->pattern().size();
```

Comment:
Might be nicer to use `s.length()` here instead, to make it clearer it is the same as what was used in allocating/copying